jerry-core/jerry.c

```diff
@@ -2055,10 +2058,17 @@ jerry_parse_and_save_snapshot (const jerry_api_char_t *source_p, /**< script sou
   size_t compiled_code_start = snapshot_buffer_write_offset;
 
   snapshot_report_byte_code_compilation = true;
+  jerry_api_object_t *error_obj_p = NULL;
 
   if (is_for_global)
   {
-    parse_status = parser_parse_script (source_p, source_size, &bytecode_data_p);
+    parse_status = parser_parse_script (source_p, source_size, &bytecode_data_p, &error_obj_p);
+
+    if (parse_status != JSP_STATUS_OK)
+    {
+      JERRY_ASSERT (error_obj_p != NULL);
+      ecma_deref_object (error_obj_p);
```

Comment:
If an unhandled exception occurs, then the error object parameter will be non-null and someone must to free it.